types/react-dom/index.d.ts

```diff
@@ -12,13 +12,25 @@ export as namespace ReactDOM;
 import {
     ReactInstance, Component, ComponentState,
     ReactElement, SFCElement, CElement,
-    DOMAttributes, DOMElement
+    DOMAttributes, DOMElement, ReactNode
 } from 'react';
 
 export function findDOMNode<E extends Element>(instance: ReactInstance): E;
 export function findDOMNode(instance: ReactInstance): Element;
 export function unmountComponentAtNode(container: Element): boolean;
 
+// @TODO: I think ReactPortal should be part of React types (as a valid ReactNode)
+// Mostly copied from Reacts own flow definitions:
+// https://github.com/facebook/react/blob/589c0a25dfa18c2090549cc6f5b626d69ea53c2a/src/renderers/shared/fiber/isomorphic/ReactTypes.js#L43-L50
+export interface ReactPortal {
```

Comment:
This should be defined in `types/react`, IMO, and should be added to the `ReactNode` type union (as you said above).